bundle/jsky.app.ot/src/main/scala/jsky/app/ot/gemini/editor/targetComponent/details/ValidAtEditor.scala

```diff
@@ -128,15 +127,16 @@ abstract class ValidAtEditor[A <: ITarget](empty: A) extends TelescopePosEditor
   ///
 
   /** A program that returns the editor's current date and time. */
-  def dateTime: HorizonsIO[Date] = {
-    val cal = Calendar.getInstance(UTC)
-    cal.setTimeZone(calendar.getTimeZone)
-    cal.setTime(calendar.getDate)
-    cal.set(Calendar.HOUR_OF_DAY, timeConfig.textDoc.getHoursField)
-    cal.set(Calendar.MINUTE,      timeConfig.textDoc.getMinutesField)
-    cal.set(Calendar.SECOND,      timeConfig.textDoc.getSecondsField)
-    HorizonsIO.delay(cal.getTime)
-  }
+  def dateTime: HorizonsIO[Date] =
+    HorizonsIO.delay {
+      val cal = Calendar.getInstance(UTC)
+      cal.setTimeZone(calendar.getTimeZone)
+      cal.setTime(calendar.getDate)
+      cal.set(Calendar.HOUR_OF_DAY, timeConfig.textDoc.getHoursField)
+      cal.set(Calendar.MINUTE, timeConfig.textDoc.getMinutesField)
+      cal.set(Calendar.SECOND, timeConfig.textDoc.getSecondsField)
+      cal.getTime
+    }
```

Comment:
I wrapped the entire thing here or else clicking on the magnifying glass would use the date at the time that the editor panel was created instead of the date displayed in the widget.

It is a bit worrisome that we now consult the widget in a non-event-loop thread and (elsewhere) update it in a non-event-loop thread.